lib/evm/instruction/impl.ex

```diff
@@ -779,7 +748,7 @@ defmodule EVM.Instruction.Impl do
   """
   @spec mstore(stack_args, vm_map) :: op_result
   def mstore([offset, value], %{machine_state: machine_state}) do
-    data = value |> mod |> :binary.encode_unsigned()
+    data = value |> wrap_int |> :binary.encode_unsigned()
```

Comment:
Should we do integer wrapping here or can we assume this will be passed pre-wrapped ints?

Reply:
Well, if we are popping values off the stack, they should have been pre-wrapped, but I don't think the extra math for verification can hurt us much.